src/Compilers/VisualBasic/Test/Semantic/IOperation/IOperationTests_ArrayCreationAndInitializer.vb

```diff
@@ -402,9 +402,12 @@ Class C
 End Class]]>.Value
 
             Dim expectedOperationTree = <![CDATA[
-IVariableDeclarationStatement (1 declarations) (OperationKind.VariableDeclarationStatement) (Syntax: 'Dim a = {}')
-  IVariableDeclaration (1 variables) (OperationKind.VariableDeclaration) (Syntax: 'a')
-    Variables: Local_1: a As System.Object()
+IVariableDeclarationGroup (1 declarations) (OperationKind.VariableDeclarationStatement) (Syntax: 'Dim a = {}')
```

Comment:
The difference between VB and C# operation trees for such single variable declarations seems to be vastly different after this change. I think this will make it extremely difficult to write a language agnostic analyzer that analyzes variable declarations for VB and C#.

Reply:
>  I think this will make it extremely difficult to write a language agnostic analyzer that analyzes variable declarations for VB and C#.

Why is that? It is easy to handle both forms.

Reply:
1. For VB, the actual initializer for the declaration is now one level above the single variable declaration which actually defines the variable.
2. `ISingleVariableDeclaration.Initializer` always has the initializer in C# (even for array declarations), but this field is not used in VB declarations, except for array bounds. I would have expected actual variable's initializer to be in this field, not array bound.

Reply:
This reflects a difference in semantics. In VB, the initializer at the declarator level actually initializes **all** variables in the declarator. The languages are simply different this way and the shape of the tree clearly reflects that. We have similar situations with "for" loops, they are just different in VB and C# and the trees are reflecting that.

Reply:
@mavasani my thinking here is that except in error scenarios, this should be fairly easy to do with the `GetDeclaredSymbols`  extension, as for most cases that + the top level initializer will be what the user is looking for. Perhaps we can add an extension method on `IVariableDeclaration` that returns an `IEnumerable` of all Initializers, something like `GetNestedInitializers`. But as Aleksey said, there's a semantic difference here that I don't think we can erase without losing information.

Reply:
I still don't like the tree shape here. I feel semantically `var x = 1` and `Dim a = 1` are identical: user is declaring a single variable with an initializer. Prior to this change, the operation trees for both these declarations indicated exactly these semantics. Now, the VB declaration is turned into a multivar declaration with 2 child nodes, the initializer `1` and a single variable declaration child with symbol `x` but no initializer - this feels like the Operation tree is trying to match how the syntax could be represented in VB, i.e. the underlying syntax tree, rather than user's intent (single variable declaration with an initializer).
The tree shape also feels weird for array creation with bounds `Dim x(1) = { }`. The single variable declaration for `x` has initializer `1`, when I would expect to see `{ }`.

I feel a better API shape here might be:
```
interface IVariableDeclarationOperation: IOperation
{
  IVariableInitializer Initializer { get; }
}

interface ISingleVariableDeclarationOperation: IVariableDeclarationOperation
{
  ISymbol DeclaredSymbol { get; }
  IOperation ArrayBound { get; }
}

interface IMultiVariableDeclarationOperation: IVariableDeclarationOperation
{
  ImmutableArray<ISingleVariableDeclarationOperation> Declarations { get; }
}
```
We use `IMultiVariableDeclarationOperation` only for declarations with 2 or more variables and `ISingleVariableDeclarationOperation` is consistently used when user declares a single variable. Initializer for the variable(s) is always the child of the operation which contains the declared symbol(s).

Also tagging @dotnet/analyzer-ioperation to get their views.